src/serialization.rs

```diff
@@ -110,10 +110,38 @@ pub mod dependency_criteria {
     }
 }
 
-/// tables with these names will be rendered by TomlFormatter as inline tables,
-/// rather than dotted tables.
-fn always_render_inline(key: &str) -> bool {
-    key == "dependency-criteria"
+/// Inline arrays which have a representation longer than this will be rendered
+/// over multiple lines.
+const ARRAY_WRAP_THRESHOLD: usize = 80;
+
+/// Tables which would be rendered inline (see `INLINE_TABLE_KEYS`) and have an
+/// inline representation larger than this threshold will be rendered by
+/// TomlFormatter as non-inline tables.
+const INLINE_TABLE_THRESHOLD: usize = 120;
+
+/// Names for tables which should be rendered inline by TomlFormatter unless
+/// their inline representation exceeds `INLINE_TABLE_THRESHOLD`.
+const INLINE_TABLE_KEYS: &[&str] = &["dependency-criteria"];
+
+fn inline_length(key: &str, value: &toml_edit::Item) -> usize {
+    // Length of the string " = " which will appear between the key and value.
+    const DECORATION_LENGTH: usize = 3;
+
+    // The `Display` implementation for toml_edit values will produce the final
+    // serialized representation of the value in the TOML document, which we can
+    // use to determine the line length in utf-8 characters.
+    let value_repr = value.to_string();
+    key.chars().count() + value_repr.chars().count() + DECORATION_LENGTH
+}
+
+/// Tables with these names will be rendered by TomlFormatter as inline tables,
+/// rather than dotted tables unless their inline representation exceeds
+/// `INLINE_TABLE_THRESHOLD` UTF-8 characters in length.
+fn render_table_inline(key: &str, value: &toml_edit::Item) -> bool {
```

Comment:
i'd prefer if this was worded more as a question, like `table_should_be_inline`. As is it sounds like it has a side-effect.